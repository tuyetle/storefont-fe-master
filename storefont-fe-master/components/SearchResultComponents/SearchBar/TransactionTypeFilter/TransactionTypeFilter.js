import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Radio from 'antd/lib/radio';
import styles from './styles';

class TransactionTypeFilter extends PureComponent {
  onChangeHandler = (evt) => {
    const {
      fieldApi: { setValue, setTouched },
    } = this.props;
    setTouched(true);
    setValue(evt.target.value);
  };

  render() {
    const RButton = Radio.Button;
    const RGroup = Radio.Group;

    const {
      options, defaultValue, size, className, ...rest
    } = this.props;

    const listOptions = options.map(option => (
      <RButton value={option.value} key={option.value} className={className}>
        {option.label}
      </RButton>));

    return (
      <RGroup
        {...rest}
        className={`${styles.radioGroupItem}`}
        defaultValue={defaultValue}
        size={size}
        onChange={this.onChangeHandler}
      >
        {listOptions}
      </RGroup>
    );
  }
}

TransactionTypeFilter.propTypes = {
  size: PropTypes.string,
  defaultValue: PropTypes.any,
  options: PropTypes.array,
  className: PropTypes.string,

  fieldApi: PropTypes.object.isRequired,
};

TransactionTypeFilter.defaultProps = {
  size: 'large',
  defaultValue: 'sell',
  options: [],
  className: '',
};

export default TransactionTypeFilter;
