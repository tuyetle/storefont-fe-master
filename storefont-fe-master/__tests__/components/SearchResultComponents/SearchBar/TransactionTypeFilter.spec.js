import React from 'react';
import renderer from 'react-test-renderer';
import TransactionTypeFilter from '~/components/SearchResultComponents/SearchBar/TransactionTypeFilter/TransactionTypeFilter';

describe('(Component) TransactionTypeFilter', () => {
  let component;
  let props;
  beforeEach(() => {
    props = {
      fieldApi: {},
    };

    component = <TransactionTypeFilter {...props} />;
  });

  it('should render conrectly', () => {
    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

