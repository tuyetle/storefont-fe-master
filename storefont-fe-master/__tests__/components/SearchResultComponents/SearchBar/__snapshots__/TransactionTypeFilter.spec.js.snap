// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) TransactionTypeFilter should render conrectly 1`] = `
<div
  className="ant-radio-group ant-radio-group-large css-19r9xb2"
  id={undefined}
  onMouseEnter={undefined}
  onMouseLeave={undefined}
  style={undefined}
/>
`;
